scripts/create_gitlab_snapshot.py

```diff
@@ -26,17 +33,26 @@
 log = logging.getLogger(__name__)
 
 
-def main():
+def main(argv):
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--no-restart',
+                        default=False,
+                        action='store_true',
+                        help='Leave the EC2 instance in a stopped state after the backup.')
+    args = parser.parse_args(argv)
     require(config.terraform_component == 'gitlab',
             "Select the 'gitlab' component ('dev.gitlab' or 'prod.gitlab', for example).")
     volume = gitlab_volume_info()
-    attachments = volume['Attachments']
-    if attachments:
-        instance = one(attachments)
+    instance = volume['Attachments']
+    if instance:
+        instance = one(instance)
         shutdown_instance(instance)
     else:
         log.info('Volume %s is not attached to any instances', volume['VolumeId'])
     create_snapshot(volume)
+    if instance and not args.no_restart:
```

Comment:
Don't like how `instance` is either an instance or an empty list here. We're ok with overwriting a variable with a value of a different type only if that overwrite is unconditional. When there are different code paths leaving the variable with differing types, it becomes confusing. 

Probably best to use [only](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.only) and undo the removal of the `attachments` variable.